doc/designs/edns.md

```diff
@@ -0,0 +1,198 @@
+| --dns-over-tls-key and --dns-over-tls-cert | in case user prefers to have the DoT certificate in BIND generated by themselves. If these are empty, IPA CA is used instead to request a new certificate. |
+| --dns-policy | Defines the DNS security policy. Accepts `relaxed` (attempts DoT but falls back to unencrypted DNS if unavailable) or `enforced` (strictly requires DoT and rejects unencrypted DNS resolution). |
+
+
+## Troubleshooting and debugging
+
+No DNS Resolution: 
+- Ensure that the DNS server (Bind or Unbound) is running.
+- Verify that the DNS server is listening on the correct IP addresses and ports.
+- Check the firewall settings to ensure that traffic on port 53 and port 853 is allowed.
+
+Encryption Not Working: 
+- Verify that forward-tls-upstream is enabled in Unbound.
+- Ensure that tls-cert-bundle or tls-system-cert is correctly configured and accessible.
+- Restart the Unbound service after making changes.
+
+Logs Not Showing Detailed Information:
+- Increase the verbosity level of logging in the configuration files for both BIND and Unbound.
+- Restart the respective services to apply the new logging levels.
+- Check the log files for detailed debug information.
+
+
+### Testing and Debugging Unbound
+
+To test the resolution from Unbound using the unbound-host standalone utility you want to use:
+
+`# unbound-host -C /etc/unbound/conf.d/tls-client.conf example.org`
+
+Then you can go through some configuration checks: ensure that Unbound is configured to use DoT automatically. Check if forward-tls-upstream was enabled and that either tls-cert-bundle or tls-system-cert was used during deployment time, as encryption will not be enabled otherwise. After making changes to the Unbound configuration, restart the Unbound daemon by running:
+
+`# systemctl restart unbound`
+
+Alternatively you can also use bind-utils to verify that resolution works through Unbound:
+
+`# dig @localhost example.org`
+
+For a very basic test of Unbound  using OpenSSL:
+
+`# openssl s_client -connect [::1]:853 -verify_hostname unbound < /dev/null`
+
+Output example:
+
+```
+CONNECTED(00000003)
+Can't use SSL_get_servername
+depth=0 CN = unbound
+verify error:num=18:self-signed certificate
+verify return:1
+depth=0 CN = unbound
+verify return:1
+---
+Certificate chain
+ 0 s:CN = unbound
+   i:CN = unbound
+   a:PKEY: rsaEncryption, 3072 (bit); sigalg: RSA-SHA256
+   v:NotBefore: Jul 13 18:09:50 2023 GMT; NotAfter: Mar 30 18:09:50 2043 GMT
+---
+```
+
+### Monitoring Traffic with tcpdump and Wireshark
+
+To watch DNS requests and ensure they are encrypted you can easily use tcpdump to capture traffic on port 53 and port 853:
+`# tcpdump -n port 53 or port 853`
+
+Alternatively you can rely on Wireshark for more detailed analysis where:
+- Port 53 queries should be visible and decodable.
+- Port 853 queries or answers should be encrypted and not decodable.
```

Comment:
Just curious, I assume the private key can be loaded into wireshark to decrypt the traffic?

Reply:
Yes, see https://wiki.wireshark.org/TLS#tls-decryption

Reply:
fixed. thank you guys.